waku/factory/node_factory.nim

```diff
@@ -223,53 +223,54 @@ proc setupProtocols(
     except CatchableError:
       return err("failed to mount waku RLN relay protocol: " & getCurrentExceptionMsg())
 
-  if conf.store and conf.legacyStore:
-    let archiveDriverRes = waitFor legacy_driver.ArchiveDriver.new(
+  if conf.store:
+    if conf.legacyStore:
+      let archiveDriverRes = waitFor legacy_driver.ArchiveDriver.new(
+        conf.storeMessageDbUrl, conf.storeMessageDbVacuum, conf.storeMessageDbMigration,
+        conf.storeMaxNumDbConnections, onFatalErrorAction,
+      )
+      if archiveDriverRes.isErr():
+        return err("failed to setup legacy archive driver: " & archiveDriverRes.error)
+
+      let retPolicyRes =
+        legacy_policy.RetentionPolicy.new(conf.storeMessageRetentionPolicy)
+      if retPolicyRes.isErr():
+        return err("failed to create retention policy: " & retPolicyRes.error)
+
+      let mountArcRes =
+        node.mountLegacyArchive(archiveDriverRes.get(), retPolicyRes.get())
+      if mountArcRes.isErr():
+        return err("failed to mount waku legacy archive protocol: " & mountArcRes.error)
+
+    ## For now we always mount the future archive driver but if the legacy one is mounted,
+    ## then the legacy will be in charge of performing the archiving.
+    ## Regarding storage, the only diff between the current/future archive driver and the legacy
+    ## one, is that the legacy stores an extra field: the id (message digest.)
+    let archiveDriverRes = waitFor driver.ArchiveDriver.new(
       conf.storeMessageDbUrl, conf.storeMessageDbVacuum, conf.storeMessageDbMigration,
       conf.storeMaxNumDbConnections, onFatalErrorAction,
     )
     if archiveDriverRes.isErr():
-      return err("failed to setup legacy archive driver: " & archiveDriverRes.error)
+      return err("failed to setup archive driver: " & archiveDriverRes.error)
 
-    let retPolicyRes =
-      legacy_policy.RetentionPolicy.new(conf.storeMessageRetentionPolicy)
+    let retPolicyRes = policy.RetentionPolicy.new(conf.storeMessageRetentionPolicy)
     if retPolicyRes.isErr():
       return err("failed to create retention policy: " & retPolicyRes.error)
 
-    let mountArcRes =
-      node.mountLegacyArchive(archiveDriverRes.get(), retPolicyRes.get())
+    let mountArcRes = node.mountArchive(archiveDriverRes.get(), retPolicyRes.get())
     if mountArcRes.isErr():
-      return err("failed to mount waku legacy archive protocol: " & mountArcRes.error)
+      return err("failed to mount waku archive protocol: " & mountArcRes.error)
 
-    # Store setup
+    # Store legacy setup
     let rateLimitSetting: RateLimitSetting =
       (conf.requestRateLimit, chronos.seconds(conf.requestRatePeriod))
-
     try:
       await mountLegacyStore(node, rateLimitSetting)
```

Comment:
Here we are calling `mountLegacyStore(node, rateLimitSetting)` even in the cases of `conf.legacyStore` being false. Is this expected?

Reply:
Thanks for that! It is corrected in https://github.com/waku-org/nwaku/pull/2894/commits/5b9871bed60420656215d3a15f28c3c7c37dcc1a